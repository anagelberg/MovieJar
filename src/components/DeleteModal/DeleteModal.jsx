import Button from "../Button/Button";
import "./DeleteModal.scss";
import ClosingX from "../ClosingX/ClosingX";
import Modal from "../Modal/Modal";

function DeleteModal({ show, closeHandler, headText, bodyText, delAction }) {

    return (
        < Modal
            show={show}
            headerContent={() => {
                return <ClosingX closeHandler={closeHandler} />
            }}

            bodyContent={() => {
                return (
                    <div className="del-modal__text">
                        <h4>{headText}</h4>
                        <p>{bodyText}</p>
                    </div>
                )
            }}

            footerContent={() => {
                return (
                    <>
                        <Button
                            text="Cancel"
                            modifier="--reverse"
                            aria-label="cancel"
                            onClick={() => closeHandler()}
                        />

                        <Button
                            text="Delete"
                            modifier="--delete"
                            aria-label="delete"
                            onClick={() => delAction()}
                        />

                    </>
                )
            }}
        />

    );
}

export default DeleteModal;