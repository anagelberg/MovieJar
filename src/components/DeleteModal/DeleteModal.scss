@use "../../styles/_mixins.scss" as *;
@use "../../styles/_variables.scss" as *;

.del-modal {
  &__close {
    fill: $dark-text;
    @include small-icon;
  }
}
