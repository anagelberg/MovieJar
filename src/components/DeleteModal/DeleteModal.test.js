import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import DeleteModal from './DeleteModal';


const mockCloseHandler = jest.fn();
const mockDelAction = jest.fn();


describe('DeleteModal Component', () => {

    beforeEach(() => {
        render(<DeleteModal show={true} closeHandler={mockCloseHandler} headText="Confirm Delete" bodyText="Are you sure you want to delete this item?" delAction={mockDelAction} />);
    });

    test('should only display when "show" prop is true', () => {
        const modal = screen.getByRole('dialog');
        expect(modal).toBeVisible();
    });

    test('should render a closingX with the correct handler', () => {
        const closeX = screen.getByTestId('closing-x');
        fireEvent.click(closeX);
        expect(mockCloseHandler).toHaveBeenCalledTimes(1);
    });

    test('should display the correct headText and bodyText', () => {
        expect(screen.getByText('Confirm Delete')).toBeInTheDocument();
        expect(screen.getByText('Are you sure you want to delete this item?')).toBeInTheDocument();
    });

    test('should render a cancel button that calls "closeHandler" when clicked', () => {
        const cancelButton = screen.getByRole('button', { name: /cancel/i });
        fireEvent.click(cancelButton);
        expect(mockCloseHandler).toHaveBeenCalledTimes(1);
    });

    test('should render a delete button that calls "delAction" when clicked', () => {
        const deleteButton = screen.getByRole('button', { name: /delete/i });
        fireEvent.click(deleteButton);
        expect(mockDelAction).toHaveBeenCalledTimes(1);
    });

    test('should have correct class modifiers for integration with buttons', () => {
        const cancelButton = screen.getByRole('button', { name: /cancel/i });
        const deleteButton = screen.getByRole('button', { name: /delete/i });
        expect(cancelButton).toHaveClass('cta--reverse');
        expect(deleteButton).toHaveClass('cta--delete');
    });

    test("matches snapshot", () => {
        const { container } = render(<DeleteModal show={true} closeHandler={mockCloseHandler} headText="Confirm Delete" bodyText="Are you sure you want to delete this item?" delAction={mockDelAction} />);
        expect(container).toMatchSnapshot();
    });
});

describe('Delete Modal Component Hidden', () => {

    test('should not display when "show" prop is false', () => {
        render(<DeleteModal show={false} closeHandler={mockCloseHandler} headText="Confirm Delete" bodyText="Are you sure you want to delete this item?" delAction={mockDelAction} />);

        expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
    });

})