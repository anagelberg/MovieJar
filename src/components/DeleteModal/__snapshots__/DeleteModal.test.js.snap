// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteModal Component matches snapshot 1`] = `
<div>
  <div
    class="modal"
    role="dialog"
  >
    <div
      class="modal__content"
    >
      <div
        class="modal__top"
      >
        <div
          class="modal__header"
        >
          <svg
            class="close-x"
            data-testid="closing-x"
          >
            close.svg
          </svg>
        </div>
        <div
          class="modal__body"
        >
          <div
            class="del-modal__text"
          >
            <h4>
              Confirm Delete
            </h4>
            <p>
              Are you sure you want to delete this item?
            </p>
          </div>
        </div>
      </div>
      <div
        class="modal__footer"
      >
        <button
          aria-label="cancel"
          class="cta cta--reverse"
        >
          <div
            class="cta__contents"
          >
            Cancel
          </div>
        </button>
        <button
          aria-label="delete"
          class="cta cta--delete"
        >
          <div
            class="cta__contents"
          >
            Delete
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
